src/Converter/ValueConverter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Webgriffe\SyliusAkeneoPlugin\Converter;
+
+use Sylius\Component\Attribute\AttributeType\SelectAttributeType;
+use Sylius\Component\Attribute\AttributeType\TextAttributeType;
+use Sylius\Component\Attribute\Model\AttributeInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class ValueConverter implements ValueConverterInterface
+{
+    /** @var TranslatorInterface|null */
+    private $translator;
+
+    /**
+     * ValueConverter constructor.
+     */
+    public function __construct(
+        TranslatorInterface $translator = null
+    ) {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convert(AttributeInterface $attribute, $value, string $localeCode)
+    {
+        if (is_array($value) && $attribute->getType() !== SelectAttributeType::TYPE) {
+            if ($attribute->getType() === TextAttributeType::TYPE) {
+                if (!array_key_exists('amount', $value)) {
+                    throw new \LogicException('Amount key not found');
+                }
+                $amount = (string) $value['amount'];
+                if (!array_key_exists('unit', $value)) {
+                    throw new \LogicException('Unit key not found');
+                }
+                $unit = (string) $value['unit'];
+                if ($this->translator === null) {
+                    return $amount . ' ' . $unit;
+                }
+
+                return $this->translator->trans('webgriffe_sylius_akeneo.ui.metric_amount_unit', ['unit' => $unit, 'amount' => $amount], null, $localeCode);
+            }
+
+            return $value;
+        }
+        if ($attribute->getType() === SelectAttributeType::TYPE && !is_bool($value)) {
+            $value = (array) $value;
+            $attributeConfiguration = $attribute->getConfiguration();
+            /** @var array $choices */
+            $choices = $attributeConfiguration['choices'];
+            $possibleOptionsCodes = array_map('strval', array_keys($choices));
+            $invalid = array_diff($value, $possibleOptionsCodes);
+
+            if (!empty($invalid)) {
+                throw new \InvalidArgumentException(
+                    sprintf(
+                        'This select attribute can only save existing attribute options. ' .
+                        'Attribute option codes [%s] do not exist.',
+                        implode(', ', $invalid)
+                    )
+                );
+            }
+
+            return [$value];
```

Comment:
Does this work for you? For me it breaks choice selection because the value has already been casted to an array.

(See https://github.com/webgriffe/SyliusAkeneoPlugin/blob/aa1406faa3d281f15594db8d1df09988862ef0ef/src/ValueHandler/AttributeValueHandler.php#L186, which is virtually identical)

Reply:
I think @aleho is right here. @lruozzi9 what do you think?

Reply:
I check right away!

Reply:
Hi @aleho. You were right. There was an error in a spec test that involved returning an array of arrays for multiple select options. It should now be correct. If you have already installed this PR I ask you to prove that the conversion is now correctly performed. Thanks again for your contribution. 😁

Reply:
Yes, that will work. I already fixed it locally by removing the return statement and importing a few hundred products yesterday. Thanks for your work!